python/pyspark/pandas/frame.py

```diff
@@ -375,6 +373,16 @@ class DataFrame(Frame, Generic[T]):
     copy : boolean, default False
         Copy data from inputs. Only affects DataFrame / 2d ndarray input
 
+    .. versionchanged:: 3.4.0
+        Since 3.4.0, it deals with `data` and `index` in this approach:
+        1, when `data` is a distributed dataset (Internal DataFrame/Spark DataFrame/
+        pandas-on-Spark DataFrame/pandas-on-Spark Series), it will first parallize
+        the `index` if necessary, and then try to combine the `data` and `index`;
+        Note that in this case `compute.ops_on_diff_frames` should be turned on;
```

Comment:
```suggestion
        1. when `data` is a distributed dataset (internal DataFrame, PySpark DataFrame,
            pandas-on-Spark DataFrame, and pandas-on-Spark Series), it will first parallelize
            the `index` if necessary, and then try to combine the `data` and `index`;
            Note that in this case `compute.ops_on_diff_frames` should be turned on;
```